### PR TITLE
Adding routine to IOutil for mkdir

### PR DESCRIPTION
**Pull request by bkochuna**
Description:
This implements a fairly portable make directory interface for
Fortran. Works for GNU and Intel.

Should work in Windows, but have not tested it there yet.

### CODE REVIEW

**Comment by zdodson94** [comment]
I had to change the syntax of the call to SYSTEM because it was causing segfaults.

**Comment by bkochuna** [comment]
I don't think the "CALL" to system works for the Intel compiler. I believe the Intel compiler has only implemented this routine as a function.

**Comment by zdodson94** [comment]
Yes, I saw that and made the fix in the vtu_org branch, which I merged mkdir into. Everything seems to work now on that branch.

File: src/IOutil.f90 [hunk 1]
```diff
@@ -244,4 +251,30 @@ SUBROUTINE GET_CURRENT_DIRECTORY_internal(dir,length,status)
...
 !
+!-------------------------------------------------------------------------------
+    SUBROUTINE MAKE_DIRECTORY_string(dir,status)
+      TYPE(StringType),INTENT(IN) :: dir
+      INTEGER(SIK),INTENT(OUT),OPTIONAL :: status
+      CALL MAKE_DIRECTORY_char(CHAR(dir),status)
+    ENDSUBROUTINE MAKE_DIRECTORY_string
+!
+!-------------------------------------------------------------------------------
+    SUBROUTINE MAKE_DIRECTORY_char(dir,status)
```
Comment (zdodson94) [inline src/IOutil.f90]: Note that "status" must be passed into this routine to avoid a segfault. Maybe it should not be optional?

Comment (youngmit) [inline src/IOutil.f90]: Just protect with `if(present(status))`. Bear in mind that present-ness gets forwarded to any subsequent calls that take optional arguments, so you only need to check `present` immediately before accessing an optional variable.

**Pull request merged**